### PR TITLE
Kats/cer-2362-edit-examples-repo-to-show-cerebrium-code-snippets

### PR DESCRIPTION
**Pull request by Katsie011**

### CODE REVIEW

**Review by elijah-rou — APPROVED** [review]
yeah looks fine

**Review by jonoirwinrsa — APPROVED** [review]

File: 1-first-cortex-deployment/config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+---
+# A config file containing all the information to deploy your model
+# You can initialise one with the `cerebrium init-cortex` command.
+cpu: 2
+disable_animation: false
+exclude: '[./.*, ./__*]'
+hardware: AMPERE_A5000
+include: '[./*, main.py, requirements.txt, pkglist.txt, conda_pkglist.txt]'
+log_level: INFO
+memory: 14.5
```
Comment (milo157) [inline 1-first-cortex-deployment/config.yaml:11]: maybe include min and max replicas and cooldown

File: 1-first-cortex-deployment/main.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from pydantic import BaseModel
+import numpy as np # import a library you've entered in your  requirements.txt
+
+class Item(BaseModel):
+    # Add your input parameters here
+    prompt: str
+    your_optional_param: Optional[str] = None # an example optional parameter
+
+### Setup your model here. 
+"""
```
Comment (milo157) [inline 1-first-cortex-deployment/main.py:11]: you must add code to this - do Stable diffusion maybe

File: 2-using-cerebrium-secrets/main.py [hunk 3]
```diff
@@ -1,24 +1,40 @@
...
+    print("\n\n")
+    print("="*60)
+    print("Error: ", e)
+    print("="*60)
+    raise e
 
-# Get input params
+huggingface_hub.login(token=hf_auth_token)
+
+########################################
```
Comment (milo157) [inline 2-using-cerebrium-secrets/main.py:35]: take out comments like this - it adds bloat and code is easy to follow

File: 2-using-cerebrium-secrets/main.py [hunk 4]
```diff
@@ -1,24 +1,40 @@
...
+    3. add the code in the line below to access your secret
+    """
+    hf_auth_token = get_secret("hf_auth_token") # load your secret
+    
+    # And that's all! It's that easy.
 
+    if hf_auth_token == "":
+        raise Exception("hf_auth_token is empty. You need a hf_auth_token secret added to your account to access this model.")
+except Exception as e:
+    print("\n\n")
```
Comment (milo157) [inline 2-using-cerebrium-secrets/main.py:27]: error messages are bloat

File: 3-using-webhook-endpoints/delay/config.yaml [hunk 5]
```diff
@@ -3,7 +3,7 @@
 cpu: 2
 disable_animation: false
 exclude: '[./.*, ./__*]'
-hardware: AMPERE_A5000
+hardware: CPU
```
Comment (milo157) [inline 3-using-webhook-endpoints/delay/config.yaml:6]: why is this in a folder called delay? Just put it in the folder normally

File: 3-using-webhook-endpoints/delay/main.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+from typing import Optional
+from pydantic import BaseModel
+import time
+
+class Item(BaseModel):
+    # Add your input parameters here
+    delayMins: Optional[int] = 3
+
+### Setup your model here. 
+"""
```
Comment (milo157) [inline 3-using-webhook-endpoints/delay/main.py:10]: remove this

File: 3-using-webhook-endpoints/delay/main.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+This code will be run once when your replica starts. 
+Load your model, setup datasets, pull in secrets, etc. here
+"""
+
+
+def predict(item, run_id, logger):
+    params = Item(**item)
+
+    for i in range(params.delayMins*10):
```
Comment (milo157) [inline 3-using-webhook-endpoints/delay/main.py:19]: remove the *10. Webhook can still be useful even if short running task. Also its just an example so doesn't need to be precise just show value

File: 3-using-webhook-endpoints/delay/main.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        logger.info(f"Waited for: {i*10} seconds...") # Some prints for your run logs
+        time.sleep(10)
+
+    return {"result":f"Delayed for {params.delayMins} mins"} # return your results 
+
+    """
+    To deploy your model, run:
+    cerebrium deploy <<YOUR DEPLOYMEN TNAME>> --config-file config.yaml
```
Comment (milo157) [inline 3-using-webhook-endpoints/delay/main.py:27]: spelling

File: 4-faster-inference-with-vllm/main.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    top_p: Optional[float] = 0.75
+    top_k: Optional[float] = 40
+    max_tokens: Optional[int] = 256
+    frequency_penalty: Optional[float] = 1
+
+
+def predict(item, run_id, logger):
+    item = Item(**item)
+
+    # Now jusst setup your sampling parameters for inference:
```
Comment (milo157) [inline 4-faster-inference-with-vllm/main.py:33]: spelling

File: 5-loading-model-weights-faster/fast_load.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
+import time
+import os
+import sys
+from transformers import AutoModelForCausalLM, AutoConfig
+
+
+# Setup the paths to the model cache
```
Comment (milo157) [inline 5-loading-model-weights-faster/fast_load.py:7]: Put link to cerebrium docs at the top of this page

File: 5-loading-model-weights-faster/fast_load.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+APP_NAME = os.environ.get("APP_NAME", "default") # this gets the name of your deployment from the environment variables
+private_model_path = f"/persistent-storage/.cache/tensorised-models/{APP_NAME}/{{org}}/{{model_name}}.tensors"
```
Comment (milo157) [inline 5-loading-model-weights-faster/fast_load.py:9]: Does the docs say why it must go here

File: 5-loading-model-weights-faster/main.py [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+    AutoTokenizer,
+    StoppingCriteria,
+    StoppingCriteriaList,
+)
+from fast_load import fast_load
+import os
+import sys
+os.environ["HUGGINGFACE_HUB_ENABLE_HF_TRANSFER"] = "1"
+
+print(("\n\n"),  file=sys. stderr)
```
Comment (milo157) [inline 5-loading-model-weights-faster/main.py:17]: these prints neccessary

File: 5-loading-model-weights-faster/main.py [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+class Item(BaseModel):
+    # Add your input parameters here
+    prompt: str
+
+# model_path = "EleutherAI/gpt-neox-20b"
+model_path = "EleutherAI/pythia-14m"
+
+
+
+#########################################################
```
Comment (milo157) [inline 5-loading-model-weights-faster/main.py:32]: take comments out

File: 5-loading-model-weights-faster/main.py [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+print(("#" * 40),  file=sys. stderr)
+print(("Starting main.py"),  file=sys. stderr)
+print(("#" * 40),  file=sys. stderr)
+print(("\n\n"),  file=sys. stderr)
+
+class Item(BaseModel):
+    # Add your input parameters here
+    prompt: str
+
+# model_path = "EleutherAI/gpt-neox-20b"
```
Comment (milo157) [inline 5-loading-model-weights-faster/main.py:27]: remove comment

File: 5-loading-model-weights-faster/main.py [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+model_path = "EleutherAI/pythia-14m"
+
+
+
+#########################################################
+# Start of the fast_load code
+#########################################################
+def setup_model():
```
Comment (milo157) [inline 5-loading-model-weights-faster/main.py:35]: why we loading twice? Users don't need to see comparison - just how to do it

File: 6-multi-gpu-inference/main.py [hunk 16]
```diff
@@ -0,0 +1,114 @@
+"""
+This main.py is a brief guide on how to implement multi-gpu using Cerebrium.
```
Comment (milo157) [inline 6-multi-gpu-inference/main.py:2]: Put docs link at top

File: 6-multi-gpu-inference/main.py [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+
+
+# Loading in base model and tokenizer
+base_model_name =  'meta-llama/Llama-2-70b-hf'  # Hugging Face Model Id
+try: 
+    hf_auth_token = get_secret("hf_auth_token")
+    if hf_auth_token == "":
+        raise Exception("hf_auth_token is empty. You need a hf_auth_token secret added to your account to access this model.")
+except Exception as e:
+    print("\n\n")
```
Comment (milo157) [inline 6-multi-gpu-inference/main.py:36]: remove

File: 6-multi-gpu-inference/main.py [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+tokenizer.pad_token_id = 0  # unk. we want this to be different from the eos token
+tokenizer.padding_side = "left"  # Allow batched inference
+
+if hasattr(base_model, "to_bettertransformer"):
+    print("Converting to BetterTransformer")
+    base_model.to_bettertransformer()
+base_model = accelerator.prepare(base_model)
+
+########################################
+# User-facing API Parameters
```
Comment (milo157) [inline 6-multi-gpu-inference/main.py:59]: remove

File: persistent-storage-management/main.py [hunk 19]
```diff
@@ -0,0 +1,86 @@
+""" 
+A little utility to view or clear files from your persistent storage.
```
Comment (milo157) [inline persistent-storage-management/main.py:2]: don't know if this is the best example. Just show saving a image or audio file or downloading weights from a s3 bucket if they don't exist

**Review by milo157 — COMMENTED** [review]
I think the 3 code folders of the examples from our website should still remain

**Pull request merged**